src/components/ModularTable/ModularTable.tsx

```diff
@@ -89,11 +93,12 @@ function ModularTable({
         })}
         {showEmpty && (
           <TableRow>
-            {/* TODO:
-              ideally this should span across whole table, but currently we don't know if any columns hides on small screens
-              so this needs to wait until we implement some centralised mechanism for hiding columns
-            */}
-            <TableCell>{emptyMsg}</TableCell>
+            <TableCell colSpan={columns.length}>{emptyMsg}</TableCell>
+          </TableRow>
+        )}
+        {footer && (
+          <TableRow>
```

Comment:
Semantically I don't think this should be here. It restricts the type of content that we can put into the footer. Can the rendering of the `footer` be moved outside of the `Table`? If for some reason it cannot then this should be in a `tfoot` but then we need to make it clear that the structure of the footer needs to fit into the defined columns of the table.

Reply:
Yes, I know it's tricky. But it's hard to get this right for all the possible (future) cases in modular table.

Right now it's just one row in the end of the table, so technically this could be in `tfoot` where I initially put it.
But as @lyubomir-popov pointed out, in future, when we have groups we may have "footer" row (and "Show more") per group. These can't be in `tfoot`, because that's always rendered below the table.

The same goes for this being outside of the table. It's meant to be consistent in positioning and styling with table rows, so we want this as one of table rows.